include/boost/geometry/extensions/random/uniform_point_distribution.hpp

```diff
@@ -0,0 +1,155 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Contributed and/or modified by Tinko Bartels,
+//   as part of Google Summer of Code 2019 program.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_UNIFORM_POINT_DISTRIBUTION_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_UNIFORM_POINT_DISTRIBUTION_HPP
+
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_envelope_rejection.hpp>
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_linear.hpp>
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_point_distribution_discrete.hpp>
+#include <boost/geometry/extensions/random/strategies/cartesian/uniform_point_distribution_box.hpp>
+#include <boost/geometry/extensions/random/strategies/spherical/uniform_inverse_transform_sampling.hpp>
+
+namespace boost { namespace geometry { namespace random
+{
+
+template
+<
+    typename DomainGeometry,
+    typename Point = typename geometry::point_type<DomainGeometry>::type,
+    typename Strategy =
+        typename strategy::uniform_point_distribution::services::default_strategy
+            <
+                Point,
+                DomainGeometry
+            >
+>
+class uniform_point_distribution
+{
+public:
+    typedef Point result_type;
+    typedef DomainGeometry domain_type;
+    class param_type {
+    public:
+        param_type(DomainGeometry const& domain) : m_domain(domain) {}
+        param_type(param_type const& p) : m_domain(p.m_domain) {}
+        DomainGeometry const& domain() const { return m_domain; }
+        bool operator==(param_type const& rhs) const
+        {
+            return equals(m_domain, rhs.m_domain);
+        }
+    private:
+        DomainGeometry m_domain;
+    };
+    uniform_point_distribution(DomainGeometry const& domain)
+        : m_strategy(domain),
```

Comment:
It should be possible to pass a CS-specific strategy into the distribution somehow. If not in the constructor then maybe in the operator(). It's because a strategy could take CS-specific parameters like radius in spherical CS or spheroid in geographic CS. This is another argument backing up the division of CS-specific part and algorithm-specific part of strategies that are currently implemented.